parachains/common/src/xcm_config.rs

```diff
@@ -119,3 +119,17 @@ impl<Location: Get<MultiLocation>> FilterAssetLocation for ConcreteNativeAssetFr
 		matches!(asset.id, Concrete(ref id) if id == origin && origin == &Location::get())
 	}
 }
+
+/// A generic function to use for MultiAssetFilter implementations, currently used to differentiate
+/// between reserve operations and the rest of them.
+pub fn weigh_multi_assets_generic(
+	filter: &MultiAssetFilter,
+	weight: Weight,
+	max_assets: u32,
+) -> XCMWeight {
+	let weight = match filter {
+		MultiAssetFilter::Definite(assets) => weight.saturating_mul(assets.len() as u64),
+		MultiAssetFilter::Wild(_) => weight.saturating_mul(max_assets as u64),
+	};
+	weight.ref_time()
```

Comment:
Could pull out the multiplier:
```suggestion
	let multiplier = match filter {
		MultiAssetFilter::Definite(assets) => assets.len() as u64,
		MultiAssetFilter::Wild(_) => max_assets as u64,
	};
	weight.saturating_mul(multiplier).ref_time()
```